src/transaction_builder.js

```diff
@@ -438,6 +448,24 @@ export class TransactionBuilder {
     return this;
   }
 
+   /**
+    * Set the {Transaction} Ext. For non-contract(non-Soroban) transactions, 
+    * Ext is defined but is unused and setting this has no effect. 
+    * In the case of Soroban transactions, Ext should be set to an instance of the 
+    * TransactionExt version 1, with SorobanTransactionData set in the instance.
+    * TransactionExt.SorobanTransactionData is obtained from a prior simulation of 
+    * the contract invocation and provides necessary resource estimations. 
+    * 
+    * @param {xdr.TransactionExt | string} ext    the Ext as xdr object or base64 string
+    * to be set as the Transaction's Ext.
+    *
+    * @returns {TransactionBuilder}
+    */
+  setExt(ext) {
```

Comment:
Maybe this is too much churn for this PR and we can do it in another iteration, but I just had a thought that might make this more ergonomic. Since the idea of an "extension" is a little unintuitive for most people, we could implicitly set `this.ext` **internally** to `v1` as needed, and instead provide `.setSorobanData(...)` method, instead?

Reply:
yes, will refactor that, saw the related reference to using it on the soroban-client pr - https://github.com/stellar/js-soroban-client/pull/77/files/4c982129767f07dbc54ccf18b347786104db2990#r1194458002